File: semana6/animales/animales.py
class Animal:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad

    def sonido(self):
        pass


class Perro(Animal):
    def sonido(self):
        return "guau"


class Gato(Animal):
    def sonido(self):
        return "miau"


class Pajaro(Animal):
    def sonido(self):
        return "pio"
animales = []

while True:
    print("Seleccione una opción:")
    print("1. Agregar Perro")
    print("2. Agregar Gato")
    print("3. Agregar Pájaro")
    print("4. Mostrar Animales")
    print("5. Salir")

    opcion = input("Elija una opción: ")

    if opcion == "1":
        nombre = input("Ingrese el nombre del perro: ")
        edad = input("Ingrese la edad del perro: ")
        perro = Perro(nombre, edad)
        animales.append(perro)

    elif opcion == "2":
        nombre = input("Ingrese el nombre del gato: ")
        edad = input("Ingrese la edad del gato: ")
        gato = Gato(nombre, edad)
        animales.append(gato)

    elif opcion == "3":
        nombre = input("Ingrese el nombre del pájaro: ")
        edad = input("Ingrese la edad del pájaro: ")
        pajaro = Pajaro(nombre, edad)
        animales.append(pajaro)

    elif opcion == "4":
        print("\nListado de Animales:")
        for animal in animales:
            print(f"Nombre: {animal.nombre}, Edad: {animal.edad}, Sonido: {animal.sonido()}")
            print("--------------------")

    elif opcion == "5":
        break

    else:
        print("Opción no válida. Por favor, elija una opción válida.")

File: semana6/animales/animales.cpp
#include <iostream>
#include <string>



class Animal {
public:
    Animal(const std::string& nombre, int edad)
        : nombre(nombre), edad(edad) {}

    virtual std::string sonido() const {
        return "";
    }

    std::string getNombre() const {
        return nombre;
    }

    int getEdad() const {
        return edad;
    }

private:
    std::string nombre;
    int edad;
};



class Perro : public Animal {
public:
    Perro(const std::string& nombre, int edad)
        : Animal(nombre, edad) {}

    std::string sonido() const override {
        return "guau";
    }
};



class Gato : public Animal {
public:
    Gato(const std::string& nombre, int edad)
        : Animal(nombre, edad) {}

    std::string sonido() const override {
        return "miau";
    }
};



class Pajaro : public Animal {
public:
    Pajaro(const std::string& nombre, int edad)
        : Animal(nombre, edad) {}

    std::string sonido() const override {
        return "pio";
    }
};